core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,515 @@
+import org.eclipse.sirius.diagram.DDiagram;
+import org.eclipse.sirius.diagram.DDiagramElement;
+import org.eclipse.sirius.diagram.DEdge;
+import org.eclipse.sirius.diagram.DNodeContainer;
+import org.eclipse.sirius.diagram.DNodeList;
+import org.eclipse.sirius.diagram.DSemanticDiagram;
+import org.eclipse.sirius.diagram.DragAndDropTarget;
+import org.eclipse.sirius.diagram.description.NodeMapping;
+import org.eclipse.sirius.viewpoint.DRepresentation;
+import org.eclipse.sirius.viewpoint.description.DAnnotation;
+import org.eclipse.sirius.viewpoint.description.DescriptionFactory;
+import org.eclipse.swt.graphics.Image;
+import org.polarsys.capella.core.data.information.impl.ClassImpl;
+import org.polarsys.capella.core.data.information.impl.DataPkgImpl;
+import org.polarsys.capella.core.sirius.analysis.activator.SiriusViewActivator;
+
+public class TitleBlockServices {
+  private static TitleBlockServices service = null;
+
+  static Map<String, String> propertiesName = new HashMap<String, String>();
+  static Map<String, String> propertiesContent = new HashMap<String, String>();
+
+  private static final String TITLE_BLOCK = "TitleBlock";
+  private static final String NAME = "Name:";
+  private static final String CONTENT = "Content:";
+  private static final String VISIBILITY = "Visibility";
+  private static final String IS_ELEMENT_TITLE_BLOCK = "Is Element Title Block";
+  private static final String TRUE = "True";
+  private static final String FALSE = "False";
+
+  public static TitleBlockServices getService() {
+    if (service == null) {
+      init();
+      service = new TitleBlockServices();
+    }
+    return service;
+  }
+
+  public static void init() {
+    propertiesName.put("TB_0_0", "Name");
+    propertiesName.put("TB_0_1", "Documentation");
+    propertiesName.put("TB_1_0", "Contextual elements");
+    propertiesContent.put("TB_0_0", "feature:name");
+    propertiesContent.put("TB_0_1", "");// "feature:documentation");
+    propertiesContent.put("TB_1_0", "");
+  }
+
+  public boolean isDiagram(EObject container) {
+    return (container instanceof DataPkgImpl);
+  }
+
+  public void createDiagramTitleBlock(EObject elementView, EObject diagram) {
+    if (elementView instanceof DSemanticDiagram && isUniqueDiagramTitleBlock(diagram)) {
+      createTitleBlock(elementView, diagram);
+    }
+  }
+
+  public void createElementTitleBlock(EObject elementView, EObject diagram) {
+    if (!(elementView instanceof DSemanticDiagram) && isUniqueElementTitleBlock(elementView, diagram)) {
+      createTitleBlock(elementView, diagram);
+    }
+  }
+
+  public boolean isDiagramTitleBlock(DAnnotation titleBlock) {
+    List<EObject> references = titleBlock.getReferences().stream().filter(x -> !(x instanceof DAnnotation))
+        .collect(Collectors.toList());
+    if (references.isEmpty()) {
+      return true;
+    }
+    return false;
+  }
+
+  private void createTitleBlock(EObject elementView, EObject diagram) {

```

Comment:
Split this function in createDiagramTitleBlock and createElementTitleBlock.

Reply:
Done.